policy/release/sbom_spdx.rego

```diff
@@ -0,0 +1,89 @@
+#
+# METADATA
+# description: >-
+#   Checks different properties of the SPDX SBOM attestation.
+#
+package policy.release.sbom_spdx
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.lib
+import data.lib.image
+
+# METADATA
+# title: Found
+# description: Confirm an SPDX SBOM attestation exists.
+# custom:
+#   short_name: found
+#   failure_msg: No SPDX SBOM attestations found
+#   solution: >-
+#     Make sure the build process produces an SPDX SBOM attestation.
+#
+deny contains result if {
+	count(_sboms) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Contains packages
+# description: Check the list of packages in the SPDX SBOM is not empty.
+# custom:
+#   short_name: contains_packages
+#   failure_msg: The list of packages is empty
+#   solution: >-
+#     Verify the SBOM is correctly identifying the package in the image.
+#
+deny contains result if {
+	some sbom in _sboms
+	count(object.get(sbom, "packages", [])) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Contains files
+# description: Check the list of files in the SPDX SBOM is not empty.
+# custom:
+#   short_name: contains_files
+#   failure_msg: The list of files is empty
+#   solution: >-
+#     Verify the SBOM is correctly identifying the files in the image.
+#
+deny contains result if {
+	some sbom in _sboms
+	count(object.get(sbom, "files", [])) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Matches image
+# description: Check the SPDX SBOM targets the image being validated.
+# custom:
+#   short_name: matches_image
+#   failure_msg: Image digest in the SBOM, %q, is not as expected, %q
+#   solution: >-
+#     The SPDX SBOM associated with the image describes a different image.
+#     Verify the integrity of the build system.
+#
+deny contains result if {
+	some sbom in _sboms
+	sbom_image := image.parse(sbom.name)
```

Comment:
Should there be a separate rule for validating the image? Can parsing `sbom.name` fail and lead to this check not happening?

Reply:
Nice catch! Added a new policy rule that verifies the SBOM predicate is valid.